### PR TITLE
Nancy testing default user agent

### PR DESCRIPTION
**Pull request by thecodejunkie**
Changes the `Nancy.Testing.Browser` to have a default `user-agent` header, with the value of `Nancy.Testing.Browser` unless explicitly set by the test code.

First reported in #1992

### CODE REVIEW

File: src/Nancy.Testing/Browser.cs [hunk 1]
```diff
@@ -199,10 +199,18 @@ public BrowserResponse Put(Url url, Action<BrowserContext> browserContext = null
...
+        /// </summary>
+        /// <param name="method">HTTP method to send the request as.</param>
+        /// <param name="url">The URl of the request.</param>
+        /// <param name="browserContext">An closure for providing browser context for the request.</param>
+        /// <returns>An <see cref="BrowserResponse"/> instance of the executed request.</returns>
         public BrowserResponse HandleRequest(string method, Url url, Action<BrowserContext> browserContext)
         {
             var request =
-                CreateRequest(method, url, browserContext ?? (with => {}));
+                this.CreateRequest(method, url, browserContext ?? (with => {}));
```
Comment (phillip-haydon) [inline src/Nancy.Testing/Browser.cs]: Why did you add `this.` on this line, but remove it on line 44 (just above)?

Comment (thecodejunkie) [inline src/Nancy.Testing/Browser.cs]: Umm not sure why it was removed on line 44.. that was not intentional! Will fix. Thanks

Comment (thecodejunkie) [inline src/Nancy.Testing/Browser.cs]: Oh sorry.. I made `DefaultBrowserContext` static. It's a method and on line 44 it's being used as a method group

Comment (phillip-haydon) [inline src/Nancy.Testing/Browser.cs]: :+1:  ok cool. I see no issues except my distaste for ternary operators. Want me to merge?
